test/integration/admin/api/backend_apis/metrics_controller_test.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class Admin::API::BackendApis::MetricsControllerTest < ActionDispatch::IntegrationTest
+  def setup
+    @tenant = FactoryBot.create(:provider_account)
+    host! @tenant.admin_domain
+    @access_token_value = FactoryBot.create(:access_token, owner: @tenant.admin_users.first!, scopes: %w[account_management], permission: 'rw').value
+    @backend_api = FactoryBot.create(:backend_api, account: @tenant)
+  end
+
+  attr_reader :backend_api, :access_token_value, :tenant
+
+  test 'index' do
+    FactoryBot.create(:metric, owner: backend_api, parent: backend_api.metrics.hits)
+
+    FactoryBot.create(:metric, owner: FactoryBot.create(:backend_api, account: tenant))
+    FactoryBot.create(:metric, service: FactoryBot.create(:service, account: tenant))
+
+    get admin_api_backend_api_metrics_path(backend_api_id: backend_api.id, access_token: access_token_value)
+
+    assert_response :success
+    assert(response_collection_metrics_of_backend_api = JSON.parse(response.body)['metrics'])
+    assert_equal 2, response_collection_metrics_of_backend_api.length
+    response_collection_metrics_of_backend_api.each do |response_metric|
+      assert backend_api.metrics.find_by(id: response_metric.dig('metric', 'id'))
+    end
+  end
+
+  test 'show' do
+    get admin_api_backend_api_metric_path(backend_api_id: backend_api.id, access_token: access_token_value, id: metric.id)
+
+    assert_response :success
+    assert_equal metric.id, JSON.parse(response.body).dig('metric', 'id')
+    assert_equal "#{metric.system_name}.#{backend_api.id}", metric.attributes['system_name']
+  end
```

Comment:
I guess we don't support "id or system_name" for the lookup, right? If we do, then we might need to add a test to find the metric by the `system_name` instead of just `id`.

Reply:
We don't support that 😄 Only `id` 💃